app/assets/javascripts/repositories/forms/repository_item_edit.js

```diff
@@ -0,0 +1,362 @@
+        $(option).removeAttr('selected');
+        if($(option).val() === selectedValue.toString()) {
+          $(option).attr('selected', true);
+          value = $(option).attr('value');
+        }
+      });
+      $(dropdown).parent().attr("list_item_id", value);
+      selectPicker.val(value);
+      selectPicker.selectpicker('refresh');
+    });
+  }
+
+  /**
+   * Creates a FormData object with the repository row data ready to be
+   * sended on the server
+   *
+   * @param {Object} tableID
+   * @param {Object} selectedRecord
+   *
+   * @returns (Object)
+   */
+  RepositoryItemEditForm.prototype.parseToFormObject = function(tableID, selectedRecord) {
+    var formData = this.formData;
+    var formDataObj = new FormData();
+    formDataObj.append('request_url', $(tableID).data('current-uri'));
+    formDataObj.append('repository_row_id', $(selectedRecord).attr('id'));
+
+    $(_.keys(this.formData)).each(function(_, element) {
+      var value = formData[element];
+      if (element === "rowName") {
+        formDataObj.append('repository_row_name', value);
+      } else {
+        var colId = element.replace('colId-', '');
+        var $el   = $('#' + element);
+        if($el.attr('type') === 'file') {
+          // don't save anything if element is deleted
+          if($el.attr('remove') === "true") {
+            return true;
+          }
+          formDataObj.append('repository_cells[' +  colId + ']',
+                             getFileValue($el));
+        } else if(value.length > 0) {
+          formDataObj.append('repository_cells[' +  colId + ']', value);
+        }
+      }
+    });
+    return formDataObj;
+  }
+  /**
+   *  |-----------------|
+   *  | Private methods |
+   *  |-----------------|
+   */
+
+   /**
+    * Resolves the file cell on FormData creation
+    *
+    * @param {Object} elementId
+    *
+    * @returns (String | Object)
+    */
+  function getFileValue(element) {
+    var file = element[0].files[0];
+    return (file) ? file : '';
+  }
+
+  /**
+   * Takes object and surrounds it with input
+   *
+   *  @param {Object} object
+   *  @param {String} name
+   *  @param {String} value
+   *  @param {String} id
+   *
+   *  @returns (String)
+   */
+  function changeToInputField(object, name, value, id) {
+    return "<div class='form-group'><input class='form-control' data-object='" +
+        object + "' name='" + name + "' value='" + value + "' id='" + id +  "'></input></div>";
+  }
+
+  /**
+   *  Takes object and creates an input file field, contains a hidden
+   *  input field which is triggered on button click and we get the uploaded
+   *  file from there.
+   *
+   *  @param {Object} object
+   *  @param {String} name
+   *  @param {String} value
+   *  @param {String} id
+   *
+   *  @returns (String)
+   */
+  function changeToInputFileField(object, name, value, id) {
+    var fileName    = (value.file_file_name) ? value.file_file_name : "";
```

Comment:
Maybe, instead of `""`, here we should also display `'No file chosen'` text (localized if possible).